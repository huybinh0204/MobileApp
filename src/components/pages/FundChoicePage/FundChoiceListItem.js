import { useTheme } from '@emotion/react';
import PropTypes from 'prop-types';
import React from 'react';
import { IconSvg } from '_components/atoms';
import { FundingOption } from './FundChoicePage.styles';

const FundChoiceListItem = ({ item }) => {
  const { colors, typography } = useTheme();

  return (
    <FundingOption
      title={item.title}
      leftItem={(props) => <IconSvg icon={item.leftItem} {...props} />}
      rightItem={(props) => <IconSvg icon={item.rightItem} {...props} />}
      onPress={item.onPress}
      testID={item.testID}
      titleStyles={{ ...typography.subContentRegular, color: colors.beta900 }}
    />
  );
};

FundChoiceListItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string.isRequired,
    leftItem: PropTypes.string,
    rightItem: PropTypes.string,
    onPress: PropTypes.func,
    testID: PropTypes.string,
  }),
};

export default FundChoiceListItem;
