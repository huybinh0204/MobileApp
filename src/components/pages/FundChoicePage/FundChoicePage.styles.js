import styled from '@emotion/native';
import { BaseText, PressableListItem } from '_components/atoms';

export const Content = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const Title = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const Description = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.m,
}));

export const FundingOption = styled(PressableListItem)(({ theme }) => ({
  backgroundColor: theme.colors.white,
  borderRadius: 10,
  borderColor: theme.colors.beta100,
  borderWidth: 1,
  marginBottom: theme.spacing.s,
  padding: theme.spacing.s,
}));

export const FundSkipNote = styled(BaseText)(({ theme }) => ({
  ...theme.typography.tinyContentRegular,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  marginTop: 'auto',
  textAlign: 'center',
}));
