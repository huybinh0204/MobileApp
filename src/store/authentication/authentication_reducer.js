import { createActions, createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';

const { Types, Creators } = createActions(
  {
    // handled by the authentication_sagas
    logout: ['isUniversalLoginEnabled'],
    signUp: ['email', 'password'],
    signIn: { email: null, password: null, isEmailVerificationEnabled: null },
    resetPassword: ['email'],
    fetchCurrentUserData: [],
    // handled by the authentication_reducer
    setAuthenticationError: ['authenticationError'],
    setCredentials: ['credentials'],
    setCurrentUserData: ['currentUserData'],
    setCurrentUserDataError: ['currentUserDataError'],
    setIsLoading: ['isLoading'],
    setIsLoadingCurrentUserData: ['isLoadingCurrentUserData'],
    setIsLoadingResetPassword: ['isLoadingResetPassword'],
    setIsSessionExpired: ['isSessionExpired'],
    setIsSignIn: ['isSignIn'],
    setResetPasswordError: ['resetPasswordError'],
    setResetPasswordSuccess: ['resetPasswordSuccess'],
    resetAuthentication: ['overrides'],
  },
  {
    prefix: '@@BE-AUTHENTICATION/',
  }
);

export const AuthenticationActions = Creators;
export const AuthenticationTypes = Types;

// INITIAL STATE
export const INITIAL_STATE = Immutable({
  authenticationError: null,
  credentials: null,
  currentUserData: null,
  currentUserDataError: null,
  isLoading: false,
  isLoadingCurrentUserData: false,
  isLoadingResetPassword: false,
  isSessionExpired: false,
  isSignIn: false,
  resetPasswordError: false,
  resetPasswordSuccess: false,
});

// SELECTORS
export const AuthenticationSelectors = {
  getAuthentication: (state) => state.auth.credentials?.accessToken,
  getAuthenticationError: (state) => state.auth.authenticationError,
  getCurrentUserData: (state) => state.auth.currentUserData,
  getCurrentUserDataError: (state) => state.auth.currentUserDataError,
  getCredentials: (state) => state.auth.credentials,
  getIsLoading: (state) => state.auth.isLoading,
  getIsLoadingCurrentUserData: (state) => state.auth.isLoadingCurrentUserData,
  getIsLoadingResetPassword: (state) => state.auth.isLoadingResetPassword,
  getIsSessionExpired: (state) => state.auth.isSessionExpired,
  getIsSignIn: (state) => state.auth.isSignIn,
  getResetPasswordError: (state) => state.auth.resetPasswordError,
  getResetPasswordSuccess: (state) => state.auth.resetPasswordSuccess,
};

// ACTION HANDLERS
const resetAuthentication = (_state, { overrides = {} }) => {
  return INITIAL_STATE.merge({ ...overrides });
};

const setAuthenticationError = (state, { authenticationError }) => {
  return state.merge({ authenticationError });
};

const setCredentials = (state, { credentials }) => {
  return state.merge({ credentials });
};

const setCurrentUserDataError = (state, { currentUserDataError }) => {
  return state.merge({ currentUserDataError });
};

const setCurrentUserData = (state, { currentUserData }) => {
  return state.merge({ currentUserData });
};

const setIsLoading = (state, { isLoading }) => {
  return state.merge({ isLoading });
};

const setIsLoadingCurrentUserData = (state, { isLoadingCurrentUserData }) => {
  return state.merge({ isLoadingCurrentUserData });
};

const setIsLoadingResetPassword = (state, { isLoadingResetPassword }) => {
  return state.merge({ isLoadingResetPassword });
};

const setIsSessionExpired = (state, { isSessionExpired }) => {
  return state.merge({ isSessionExpired });
};

const setIsSignIn = (state, { isSignIn }) => {
  return state.merge({ isSignIn });
};

const setResetPasswordError = (state, { resetPasswordError }) => {
  return state.merge({ resetPasswordError });
};

const setResetPasswordSuccess = (state, { resetPasswordSuccess }) => {
  return state.merge({ resetPasswordSuccess });
};

// REDUCER
export const AuthenticationReducer = createReducer(INITIAL_STATE, {
  [Types.SET_AUTHENTICATION_ERROR]: setAuthenticationError,
  [Types.SET_CREDENTIALS]: setCredentials,
  [Types.SET_CURRENT_USER_DATA]: setCurrentUserData,
  [Types.SET_CURRENT_USER_DATA_ERROR]: setCurrentUserDataError,
  [Types.SET_IS_LOADING]: setIsLoading,
  [Types.SET_IS_LOADING_CURRENT_USER_DATA]: setIsLoadingCurrentUserData,
  [Types.SET_IS_LOADING_RESET_PASSWORD]: setIsLoadingResetPassword,
  [Types.SET_IS_SESSION_EXPIRED]: setIsSessionExpired,
  [Types.SET_IS_SIGN_IN]: setIsSignIn,
  [Types.SET_RESET_PASSWORD_ERROR]: setResetPasswordError,
  [Types.SET_RESET_PASSWORD_SUCCESS]: setResetPasswordSuccess,
  [Types.RESET_AUTHENTICATION]: resetAuthentication,
});
