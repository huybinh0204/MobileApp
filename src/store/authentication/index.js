export * from './authentication_reducer';
export * from './authentication_sagas';
