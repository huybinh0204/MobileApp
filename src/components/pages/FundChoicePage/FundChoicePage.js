import { useNavigation } from '@react-navigation/native';
import React, { useEffect, useMemo } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { MainButton } from '_components/atoms';
import { OnboardingScreenLayout } from '_components/organisms';
import { FEATURE_FLAGS, ICONS, NAVIGATION } from '_constants';
import { useBooleanFeatureFlag } from '_hooks';
import strings from '_localization';
import { AccountActions, AccountSelectors } from '_store/account';
import { AuthenticationActions, AuthenticationSelectors } from '_store/authentication';
import { CustomerSelectors } from '_store/customer';
import { FundDirectDepositStoreActions } from '_store/pageStore/fundDirectDepositStore';
import FundChoiceListItem from './FundChoiceListItem';
import { Content, Description, FundSkipNote, Title } from './FundChoicePage.styles';

const FundChoicePage = () => {
  const dispatch = useDispatch();
  const { navigate } = useNavigation();

  const accountExternalId = useSelector(AccountSelectors.getExternalId);
  const customerExternalId = useSelector(CustomerSelectors.getExternalId);
  const credentials = useSelector(AuthenticationSelectors.getCredentials);

  const isAchInEnabled = useBooleanFeatureFlag(FEATURE_FLAGS.ACH_IN, customerExternalId);

  const fundingOptions = useMemo(() => {
    return [
      {
        enabled: true,
        title: strings.signUp.fund.directDeposit,
        leftItem: ICONS.directDeposit,
        rightItem: ICONS.arrowRightActive,
        onPress() {
          navigate(NAVIGATION.shared.earlyPaycheck);
        },
      },
      {
        enabled: isAchInEnabled,
        title: strings.signUp.fund.bankTransfer,
        leftItem: ICONS.bankCircle,
        rightItem: ICONS.arrowRightActive,
        onPress() {
          navigate(NAVIGATION.shared.addMoney);
        },
      },
      {
        enabled: true,
        title: strings.signUp.fund.depositCash,
        leftItem: ICONS.depositCash,
        rightItem: ICONS.arrowRightActive,
        onPress() {
          navigate(NAVIGATION.auth.depositCash);
        },
      },
    ];
  }, [isAchInEnabled, navigate]);

  const handleSkip = () => {
    dispatch(AuthenticationActions.setIsSignIn(true));
  };

  useEffect(() => {
    dispatch(FundDirectDepositStoreActions.fetchRenderData(accountExternalId));
    dispatch(AccountActions.fetchPlaidLinkToken(credentials?.sub));
  }, [dispatch, accountExternalId, credentials?.sub]);

  return (
    <OnboardingScreenLayout step={10} testID="FundChoicePage">
      <Content>
        <Title>{strings.signUp.fund.title}</Title>
        <Description>{strings.signUp.fund.content}</Description>
        {fundingOptions.map((option) =>
          option.enabled ? <FundChoiceListItem item={option} key={option.title} /> : null
        )}
        <FundSkipNote>{strings.signUp.fund.skipNote}</FundSkipNote>
        <MainButton accessibilityLabel="SkipFundChoiceButton" onPress={handleSkip} variant="link">
          {strings.signUp.fund.skipButton}
        </MainButton>
      </Content>
    </OnboardingScreenLayout>
  );
};

export default FundChoicePage;
