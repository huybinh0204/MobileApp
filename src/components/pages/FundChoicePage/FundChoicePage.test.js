import { useNavigation } from '@react-navigation/native';
import * as Split from '@splitsoftware/splitio-react';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { NAVIGATION } from '_constants';
import strings from '_localization/index';
import { AuthenticationActions } from '_store/authentication';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import FundChoicePage from './FundChoicePage';

const FundChoise = () => (
  <Split.SplitContext.Provider value={{ isReady: true }}>
    <FundChoicePage />
  </Split.SplitContext.Provider>
);

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('FundChoicePage', () => {
  const mockDispatch = jest.fn();
  const mockNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useDispatch.mockReturnValue(mockDispatch);
    useNavigation.mockReturnValue(mockNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    jest.spyOn(Split, 'useClient').mockReturnValue({
      getTreatment: () => 'on',
    });

    const { toJSON } = render(<ComponentWithProviders component={FundChoise} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to the home tab when skip button is clicked', () => {
    const { getByText } = render(<ComponentWithProviders component={FundChoise} />);

    const skipButton = getByText(strings.signUp.fund.skipButton);

    fireEvent.press(skipButton);

    expect(mockDispatch).toHaveBeenCalledWith(AuthenticationActions.setIsSignIn(true));
  });

  it('should navigate to direct deposit info page', () => {
    const { getByText } = render(<ComponentWithProviders component={FundChoise} />);

    const directDepositButton = getByText(strings.signUp.fund.directDeposit);

    fireEvent.press(directDepositButton);

    expect(mockNavigation.navigate).toHaveBeenCalledWith(NAVIGATION.shared.earlyPaycheck);
  });

  it('should navigate to fund amount from external bank page when ACH In is enabled', () => {
    jest.spyOn(Split, 'useClient').mockReturnValue({
      getTreatment: () => 'on',
    });

    const { getByText } = render(<ComponentWithProviders component={FundChoise} />);

    const addMoney = getByText(strings.signUp.fund.bankTransfer);

    fireEvent.press(addMoney);

    expect(mockNavigation.navigate).toHaveBeenCalledWith(NAVIGATION.shared.addMoney);
  });

  it('should not show the transfer from external bank option when ACH In is disabled', () => {
    jest.spyOn(Split, 'useClient').mockReturnValue({
      getTreatment: () => 'off',
    });

    const { queryByText } = render(<ComponentWithProviders component={FundChoise} />);

    const achInOption = queryByText(strings.signUp.fund.bankTransfer);

    expect(achInOption).toBeNull();
  });

  it('should navigate to deposit cash page', () => {
    const { getByText } = render(<ComponentWithProviders component={FundChoise} />);

    const depositCash = getByText(strings.signUp.fund.depositCash);

    fireEvent.press(depositCash);

    expect(mockNavigation.navigate).toHaveBeenCalledWith(NAVIGATION.auth.depositCash);
  });
});
