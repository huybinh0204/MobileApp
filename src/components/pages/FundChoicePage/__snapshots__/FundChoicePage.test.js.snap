// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FundChoicePage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="FundChoicePage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "width": "50%",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "backgroundColor": "#F2F3F4",
                                    "borderColor": "#000000",
                                    "borderRadius": 10,
                                    "height": 9,
                                    "width": "100%",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "bottom": 0,
                                    "left": 0,
                                    "position": "absolute",
                                    "right": 0,
                                    "top": 0,
                                    "width": "0%",
                                  }
                                }
                              />
                            </View>
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 36,
                                    "lineHeight": 50.5,
                                    "marginBottom": 14.5,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Fund your account
                            </Text>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "marginBottom": 58,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Deposit money into your new account to get ahead with your money.
                            </Text>
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#FFFFFF",
                                  "borderColor": "#D4D7DB",
                                  "borderRadius": 10,
                                  "borderWidth": 1,
                                  "flexDirection": "row",
                                  "marginBottom": 29,
                                  "minHeight": 50,
                                  "opacity": 1,
                                  "padding": 29,
                                }
                              }
                              testID="PressableListItemContainer"
                            >
                              <SvgMock
                                height={58}
                                testID="LeftItem"
                                width={58}
                              />
                              <Text
                                allowFontScaling={false}
                                marginLeft={true}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "marginLeft": 29,
                                    },
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                    },
                                  ]
                                }
                                testID="ItemTitle"
                              >
                                Add Direct Deposit
                              </Text>
                              <SvgMock
                                height={45.5}
                                testID="RightItem"
                                width={45.5}
                              />
                            </View>
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#FFFFFF",
                                  "borderColor": "#D4D7DB",
                                  "borderRadius": 10,
                                  "borderWidth": 1,
                                  "flexDirection": "row",
                                  "marginBottom": 29,
                                  "minHeight": 50,
                                  "opacity": 1,
                                  "padding": 29,
                                }
                              }
                              testID="PressableListItemContainer"
                            >
                              <SvgMock
                                height={58}
                                testID="LeftItem"
                                width={58}
                              />
                              <Text
                                allowFontScaling={false}
                                marginLeft={true}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "marginLeft": 29,
                                    },
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                    },
                                  ]
                                }
                                testID="ItemTitle"
                              >
                                Bank transfer
                              </Text>
                              <SvgMock
                                height={45.5}
                                testID="RightItem"
                                width={45.5}
                              />
                            </View>
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "backgroundColor": "#FFFFFF",
                                  "borderColor": "#D4D7DB",
                                  "borderRadius": 10,
                                  "borderWidth": 1,
                                  "flexDirection": "row",
                                  "marginBottom": 29,
                                  "minHeight": 50,
                                  "opacity": 1,
                                  "padding": 29,
                                }
                              }
                              testID="PressableListItemContainer"
                            >
                              <SvgMock
                                height={58}
                                testID="LeftItem"
                                width={58}
                              />
                              <Text
                                allowFontScaling={false}
                                marginLeft={true}
                                style={
                                  Array [
                                    Object {
                                      "flex": 1,
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "marginLeft": 29,
                                    },
                                    Object {
                                      "color": "#1F1D25",
                                      "fontFamily": "Poppins",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                    },
                                  ]
                                }
                                testID="ItemTitle"
                              >
                                Deposit cash
                              </Text>
                              <SvgMock
                                height={45.5}
                                testID="RightItem"
                                width={45.5}
                              />
                            </View>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 21.5,
                                    "lineHeight": 30.5,
                                    "marginBottom": 29,
                                    "marginTop": "auto",
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              You can always add funds later. 
*Additional terms apply. See Direct Deposit terms.
                            </Text>
                            <View
                              accessibilityLabel="SkipFundChoiceButton"
                              accessibilityState={
                                Object {
                                  "disabled": false,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "alignSelf": "stretch",
                                  "backgroundColor": "transparent",
                                  "borderRadius": 10,
                                  "opacity": 1,
                                  "paddingHorizontal": 45.5,
                                  "paddingVertical": 31,
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                disabled={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#3D3397",
                                      "fontFamily": "Poppins-SemiBold",
                                      "fontSize": 25.5,
                                      "lineHeight": 35.5,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                Skip
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
